backbone.blazer.js

```diff
@@ -60,19 +65,19 @@ Backbone.Blazer.Router = Backbone.Router.extend({
         route.trigger('before:execute', routeData);
         router.trigger('before:execute', routeData);
 
-        this._runBeforeFilters(route, routeData).then(function() {
-            return $.when(route.prepare(routeData));
-        }).then(function() {
+        this._runBeforeFilters(router, route, routeData).then(function() {
+            return this._runHandler(route.prepare, router, route, routeData);
+        }.bind(this)).then(function() {
```

Comment:
yeah, we're using the `router` alias for `this` in other parts of these callbacks already too